explorer/data/prelude.carbon

```diff
@@ -156,6 +156,10 @@ interface GreaterEqWith(U:! Type) {
   fn GreaterEq[me: Self](other: U) -> bool;
 }
 
+interface NotEqWith(U:! Type){
+  fn NotEq[me: Self](other: U) -> bool;
```

Comment:
This doesn't match the [current design](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/design/expressions/comparison_operators.md), where the `NotEqual` method is part of the same interface as `Equal`.

Reply:
I'd agreed to doing this for the various comparisons, which by design, belong on an OrderedWith interface. The problem is that `default fn` isn't implemented, but people do want to use the features. I think the incremental cost of accepting changes such as this one is pretty low, and we can clean up later, so my own opinion is to accept things like this with a TODO indicating the necessary cleanup.

Reply:
Sorry... I had the design paper not longer in my mind. You are complety right that is  unnecessarily verbose.

Reply:
> I'd agreed to doing this for the various comparisons, which by design, belong on an OrderedWith interface. The problem is that `default fn` isn't implemented, but people do want to use the features.

I see. In that case I'm OK either way, but I prefer the way @pmqtt has now done it, which matches the design but with `default` omitted. Are you OK with this as it is?